### PR TITLE
feat: Realtime HTTP APIs/Websocket messages structures v1

### PR DESCRIPTION
**Pull request by himank**

### CODE REVIEW

File: server/v1/realtime.proto [hunk 1]
```diff
@@ -0,0 +1,241 @@
...
+message UnSubscribeResponse {
+  string status=1;
+}
+
+message ListSubscriptionRequest {
+  string project=1;
+  string channel=2;
+}
+
+message ListSubscriptionResponse {
```
Comment (adilansari) [inline server/v1/realtime.proto:175]: this might need some sort of pagination for HTTP

**Review by adilansari — DISMISSED** [review]

File: server/v1/realtime.proto [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+    option (google.api.http) = {
+      get : "/v1/projects/{project}/realtime/channels/{channel}/presence"
+    };
+    option(openapi.v3.operation) = {
+      summary: "Presence about the channel"
+      tags: "Realtime"
+      security: []
+    };
+  }
+  rpc GetChannel(GetChannelRequest) returns (GetChannelResponse) {
```
Comment (garrensmith) [inline server/v1/realtime.proto:190]: we will have to rename `GetChannel` and `GetChannels` methods. The underlying client in Typescript already has those two methods for internal usage so we get a typescript error and would get a runtime error as well:

```
roperty 'getChannel' in type 'RealtimeClient' is not assignable to the same property in base type 'Client'.
  Type '{ (request: GetChannelRequest, callback: (error: ServiceError, response: GetChannelResponse) => void): SurfaceCall; (request: GetChannelRequest, metadata: Metadata, callback: (error: ServiceError, response: GetChannelResponse) => void): SurfaceCall; (request: GetChannelRequest, metadata: Metadata, options: Partial<....' is not assignable to type '() => Channel'.

111     public getChannel(request: server_v1_realtime_pb.GetChannelRequest, callback: (error: grpc.ServiceError | null, response: server_v1_realtime_pb.GetChannelResponse) => void): grpc.ClientUnaryCall;
```

Maybe something like `GetTigrisChannel` or `GetMsgChannel` or `GetRTChannel`

**Review by garrensmith — CHANGES_REQUESTED** [review]
See my note on the channel rpc call. It breaks when trying to use it in the js client

File: server/v1/realtime.proto (hunk 1 above)
Comment (himank) [reply server/v1/realtime.proto:175]: ack, let me add limit and offset for now.

Comment (himank) [reply server/v1/realtime.proto:177]: Or page similar to Search APIs.

**Review by efirs — APPROVED** [review]

File: server/v1/realtime.proto (hunk 1 above)
Comment (adilansari) [reply server/v1/realtime.proto:177]: We should add total subscriptions and total pages here as well. That helps bottom-up traversal, or why suddenly pagination stopped returning results etc.

**Review by adilansari — DISMISSED** [review]
we can add suggestions later

File: server/v1/realtime.proto (hunk 1 above)
Comment (himank) [reply server/v1/realtime.proto:177]: sure, once the response is finalized will add it.

**Review by garrensmith — APPROVED** [review]
Tested in js client and it now passes compile stage.

File: server/v1/realtime.proto [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+}
+
+message ConnectedMessage {
+  string session_id=1;
+  string socket_id=2;
+  string event=3;
+}
+
+message RealTimeMessage {
+  string action=1;
```
Comment (garrensmith) [inline server/v1/realtime.proto:54]: Lets remove action and instead add in new event types for `presence`.

File: server/v1/realtime.proto [hunk 4]
```diff
@@ -0,0 +1,243 @@
...
+  string event_type=2;
```
Comment (garrensmith) [inline server/v1/realtime.proto:55]: Can we make this an enum

File: server/v1/realtime.proto [hunk 5]
```diff
@@ -0,0 +1,243 @@
...
+  string position=2;
+  // name here is what type of event that the caller is interested in
+  string name = 3;
+}
+
+message MessageEvent {
+  string id=1;
+  string name=2;
+  string channel=3;
+  string timestamp=4;
```
Comment (garrensmith) [inline server/v1/realtime.proto:90]: lets remove timestamp for now and we can add it back later if we need it. 
Add in a `seq` instead that shows the position of the message in the channel stream

Comment (himank) [reply server/v1/realtime.proto]: actually `id` is the sequence, I completely missed it.  So there is no separate `seq` field needed.

File: server/v1/realtime.proto (hunk 4 above)
Comment (himank) [reply server/v1/realtime.proto]: Yes, it is an enum now. We need to also publish these enum codes for HTTPs.

**Review by garrensmith — APPROVED** [review]

**Pull request merged**